src/lib/crypto/kimchi_bindings/stubs/src/lagrange_basis.rs

```diff
@@ -0,0 +1,204 @@
+use ark_ec::AffineCurve;
+use ark_poly::{EvaluationDomain, Radix2EvaluationDomain as D};
+use cache::LagrangeCache;
+use mina_curves::pasta::{Pallas, Vesta};
+use poly_commitment::{commitment::CommitmentCurve, srs::SRS};
+use std::env;
+
+pub trait WithLagrangeBasis<G: AffineCurve> {
+    fn with_lagrange_basis(&mut self, domain: D<G::ScalarField>);
+}
+
+impl WithLagrangeBasis<Vesta> for SRS<Vesta> {
+    fn with_lagrange_basis(&mut self, domain: D<<Vesta as AffineCurve>::ScalarField>) {
+        let use_cache =
+            env::var("USE_LAGRANGE_CACHE").unwrap_or_else(|_| "false".to_string()) == "true";
+        if use_cache {
+            add_lagrange_basis_with_cache(self, domain, cache::get_vesta_file_cache());
+        } else {
+            println!("NOT USING LAGRANGE CACHE");
```

Comment:
Probably want to delete this so that it doesn't print every time you prove without the env var (e.g. for the Mina node!)